assets/javascripts/modules/checkout/fieldSwitcher.js

```diff
@@ -4,63 +4,127 @@ define([
+
+    var updateLocalization = function (currentState) {
+        var updateParams = {};
+
+        var addFieldsFor = function (addressType) {
+            var prefix = addressType == 'localization' ? '' : addressType + '-';
+            var values = currentState[addressType];
+            if (!isEmptyObject(values)) {
+                if (values.currency.length > 0) {
+                    updateParams[prefix + 'currency'] = values.currency;
+                }
+                if (values.country.length > 0) {
+                    updateParams[prefix + 'country'] = values.country;
+                }
+            }
+        };
+
+        addFieldsFor('delivery');
+        addFieldsFor('billing');
+        addFieldsFor('localization');
+        var selectedPlanBeforeUpdate = ratePlanChoice.getSelectedRatePlanId();
+        dataSwitcher.refresh(updateParams);
+        checkPlanInput(selectedPlanBeforeUpdate, currentState.localization.currency);
+        refreshPaymentMethods(currentState);
+    };
+
+    var getCurrentState = function () {
+        var deliveryState = addressData.delivery.getState();
+        var billingState = deliveryAsBilling.isEnabled() ? deliveryState : addressData.billing.getState();
+        var localizationState = billingState.determinesLocalization ? billingState : deliveryState;
+
+        if (isCurrencyOverrideChecked()) {
+            localizationState.currency = 'GBP';
+        }
+        return {
+            delivery: deliveryState,
+            billing: billingState,
+            localization: localizationState
+        }
+    };
+    var update = function () {
+        var currentState = getCurrentState();
+        updateLocalization(currentState)
+    };
+
+    var isCurrencyOverrideChecked = function() {
+        var currencyOverrudeCheckbox = $('.js-currency-override-checkbox');
+        return currencyOverrudeCheckbox.length >0 && currencyOverrudeCheckbox[0].checked;
+
     };
+
     var initCurrencyOverride = function () {
-        var currencyOverrideCheckbox = $('.js-currency-override-checkbox')[0];
-        bean.on(currencyOverrideCheckbox, 'change', function () {
-            if (currencyOverrideCheckbox.checked) {
-                overrideCurrency('GBP');
-            }
-            else {
-                overrideCurrency('USD');
-            }
+        $('.js-currency-override-checkbox').each(function (currencyOverrideCheckbox) {
+            bean.on(currencyOverrideCheckbox, 'change', function () {
+              update();
+            });
         });
     };
-    var overrideCurrency = function(currency) {
-        latestCurrencyAddressModel.currency = currency;
-        switchCurrency(latestCurrencyAddressModel);
+
+    var redrawCurrencyOverride = function (currentState) {
+        $('.js-currency-override-checkbox').attr('checked', false);
+
+        var currencySelector = $('.js-currency-override-label, .js-currency-selector');
+
+        if (currentState.localization.currency == 'USD') {
```

Comment:
Note, way may change this to if (currentState.localization.currency != 'GBP') in future if we want to allow AUS, NZA and EUR customers to pay in GBP too. CAD customers are already currently excluded due to being in the other zone.